nav2_controller/plugins/feasible_path_handler.cpp

```diff
@@ -0,0 +1,291 @@
+  if (!nav2_util::transformPoseInTargetFrame(pose, robot_pose, *tf_,
+      global_plan_up_to_constraint_.header.frame_id,
+      transform_tolerance_))
+  {
+    throw nav2_core::ControllerTFError("Unable to transform robot pose into global plan's frame");
+  }
+
+  return robot_pose;
+}
+
+nav2_core::PathSegment FeasiblePathHandler::findPlanSegment(
+  const geometry_msgs::msg::PoseStamped & pose)
+{
+  global_pose_ = transformToGlobalPlanFrame(pose);
+  // Limit the search for the closest pose up to max_robot_pose_search_dist on the path
+  auto closest_pose_upper_bound =
+    nav2_util::geometry_utils::first_after_integrated_distance(
+    global_plan_up_to_constraint_.poses.begin(), global_plan_up_to_constraint_.poses.end(),
+      max_robot_pose_search_dist_);
+
+  // First find the closest pose on the path to the robot
+  // bounded by when the path turns around (if it does) so we don't get a pose from a later
+  // portion of the path
+  auto closest_point =
+    nav2_util::geometry_utils::min_by(
+    global_plan_up_to_constraint_.poses.begin(), closest_pose_upper_bound,
+    [this](const geometry_msgs::msg::PoseStamped & ps) {
+      return euclidean_distance(global_pose_, ps);
+    });
+
+  // Make sure we always have at least 2 points on the transformed plan and that we don't prune
+  // the global plan below 2 points in order to have always enough point to interpolate the
+  // end of path direction
+  if (global_plan_up_to_constraint_.poses.begin() != closest_pose_upper_bound &&
+    global_plan_up_to_constraint_.poses.size() > 1 &&
+    closest_point == std::prev(closest_pose_upper_bound))
+  {
+    closest_point = std::prev(std::prev(closest_pose_upper_bound));
+  }
+
+  auto pruned_plan_end =
+    nav2_util::geometry_utils::first_after_integrated_distance(
+    closest_point, global_plan_up_to_constraint_.poses.end(), prune_distance_);
+
+  return {closest_point, pruned_plan_end};
+}
+
+
+nav_msgs::msg::Path FeasiblePathHandler::transformLocalPlan(
+  const nav2_core::PathIterator & closest_point,
+  const nav2_core::PathIterator & pruned_plan_end)
+{
+  nav_msgs::msg::Path transformed_plan;
+  transformed_plan.header.frame_id = costmap_ros_->getGlobalFrameID();
+  transformed_plan.header.stamp = global_pose_.header.stamp;
+  unsigned int mx, my;
+  // Find the furthest relevant pose on the path to consider within costmap
+  // bounds
+  // Transforming it to the costmap frame in the same loop
+  for (auto global_plan_pose = closest_point; global_plan_pose != pruned_plan_end;
+    ++global_plan_pose)
+  {
+    // Transform from global plan frame to costmap frame
+    geometry_msgs::msg::PoseStamped costmap_plan_pose;
+    global_plan_pose->header.stamp = global_pose_.header.stamp;
+    global_plan_pose->header.frame_id = global_plan_.header.frame_id;
+    nav2_util::transformPoseInTargetFrame(*global_plan_pose, costmap_plan_pose, *tf_,
+        costmap_ros_->getGlobalFrameID(), transform_tolerance_);
```

Comment:
```suggestion
      costmap_ros_->getGlobalFrameID(), transform_tolerance_);
```

I feel like something must have changed recently in vscode. I'm getting alot of PRs from alot of people with mixed in 4 spaces that I cannot explain. Its not just you, its everyone :(